core/src/main/java/io/opencensus/tags/TagValueString.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2017, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.tags;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.base.Preconditions;
+import io.opencensus.internal.StringUtil;
+import io.opencensus.tags.TagKey.TagKeyString;
+import javax.annotation.concurrent.Immutable;
+
+/** A validated tag value associated with a {@link TagKeyString}. */
+@Immutable
+@AutoValue
+public abstract class TagValueString {
```

Comment:
We really don't need (so far) a TagValue class that is TagValueString | TagValueLong | TagValueBoolean (the Tag class obviates that).

This particular class is meant to be restriction on String values - it may make more sense to rename it something like CensusString for restrictions on String's that are enforced Census-wide (e.g. printable ascii characters only) but we need to agree Census-wide, what such restrictions are.